dialects/postgresql/src/main/kotlin/app/cash/sqldelight/dialects/postgresql/grammar/mixins/RegExMatchOperatorExpressionMixin.kt

```diff
@@ -0,0 +1,31 @@
+package app.cash.sqldelight.dialects.postgresql.grammar.mixins
+
+import app.cash.sqldelight.dialects.postgresql.grammar.psi.PostgreSqlRegexMatchOperatorExpression
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.SqlBinaryExpr
+import com.alecstrong.sql.psi.core.psi.SqlColumnDef
+import com.alecstrong.sql.psi.core.psi.SqlColumnName
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElementImpl
+import com.alecstrong.sql.psi.core.psi.SqlExpr
+import com.intellij.lang.ASTNode
+
+/**
+ * Regular expression operators provide a more powerful means for pattern matching than the LIKE and SIMILAR TO operators.
+ */
+internal abstract class RegExMatchOperatorExpressionMixin(node: ASTNode) :
+  SqlCompositeElementImpl(node),
+  SqlBinaryExpr,
+  PostgreSqlRegexMatchOperatorExpression {
+
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
+    val columnType = ((firstChild.firstChild.reference?.resolve() as? SqlColumnName)?.parent as? SqlColumnDef)?.columnType?.typeName?.text
```

Comment:
would using the type resolver here work?

```
val columnType = sqFile().typeResolver.resolvedType(getExprList()[0]).dialectType == PrimitiveType.TEXT
```

then the annotator will also work if a function doesn't return a string

Reply:
😞 It seems that all the typeResolver  api needed is in the core compiler api and this is not available in the "frontend" api dialect 

Validating functions - one returns text and the other returns int - is not possible - so I currently allow it
```sql
SELECT concat(t, 'test') ~ ? 
FROM regexops;

SELECT length(t) ~ ?  
FROM regexops;
```